### ISSUE TITLE
ch-run.c:205: 2: No such file or directory

### ISSUE THREAD

**Issue by AdamNSmith** [hpc/charliecloud#1]
When I try to start /bin/bash in a Centos 6.8 image using the command: `ch-run centos6/ /bin/bash`, ch-run errors with the message "ch-run.c:205: 2: No such file or directory". I found that if I un-comment line #204 and comment out line #205, so that the section is:

```
203    // Post pivot_root() tmpfs
204    TRY (mount(NULL, "/dev/shm", "tmpfs", 0, "size=4m"));  // for CLONE_NEWIPC
205    //TRY (mount(NULL, "/run", "tmpfs", 0, "size=10%"));
```
then everything works correctly.

**Comment by reidpr** [comment]
I believe this happens because CentOS doesn't have `/run` but rather `/var/run`. In contrast, Debian has a `tmpfs` on `/run`, and `/var/run` is a symlink to `/run`.

I think the fix might be to try both and use whichever is a directory.

**Comment by reidpr** [comment]
I haven't fixed it yet — re-opening.

**Comment by reidpr** [comment]
On second thought, this isn't limited to `/run` and `/var/run`; there are a number of directories that an app might want to write to, and enumerating them to put a `tmpfs` on each doesn't seem like the right solution.

I did try `overlayfs` to make the image writeable, but this can't be used with user namespaces in the upstream 4.4.

One solution is to simply leave it up to the packager, by configuring the app or making some symlinks. I added a FAQ entry to this effect, which will be pushed sometime soon.

Closing. If this is isn't the right solution, we can re-open or create a new issue.
